.github/workflows/go.yml

```diff
@@ -0,0 +1,186 @@
+name: Go CI
+
+on:
+    workflow_dispatch:
+    push:
+        branches: [ "main" ]
+        paths:
+            - glide-core/**
+            - submodules/**
+            - go/**
+            - .github/workflows/go.yml
+    pull_request:
+        paths:
+            - glide-core/**
+            - submodules/**
+            - go/**
+            - .github/workflows/go.yml
+
+# Run only the latest job on a branch and cancel previous ones
+concurrency:
+    group: ${{ github.head_ref || github.ref }}
+    cancel-in-progress: true
+
+jobs:
+    build-and-test-go-client:
+        timeout-minutes: 10
+        strategy:
+            # Run all jobs
+            fail-fast: false
+            matrix:
+                go:
+                    - '1.18'
+                    - '1.21'
+                redis:
+                    - 6.2.14
+                    - 7.2.3
+                os:
+                    - ubuntu-latest
+                    - macos-latest
+
+        runs-on: ${{ matrix.os }}
+
+        steps:
+            -   uses: actions/checkout@v4
+                with:
+                    submodules: recursive
+
+            -   name: Set up Go ${{ matrix.go }}
+                uses: actions/setup-go@v5
```

Comment:
Please try using `cache-dependency-path` to speed up the build and suppress the [warning about missing cache](https://github.com/aws/glide-for-redis/actions/runs/7894170720?pr=946)
ref: https://github.com/actions/setup-go#caching-dependency-files-and-build-outputs